Common/src/CConfig.cpp

```diff
@@ -6008,6 +6009,35 @@ void CConfig::SetMarkers(SU2_COMPONENT val_software) {
     Marker_CfgFile_MixingPlaneInterface[iMarker_CfgFile] = indexMarker;
   }
 
+  /*--- Once we have identified the MixingPlane and Turbomachinery markers
+   *    we next need to determine what type of interface between zones is
+   *    used. It is convenient to do this here as it tidies up the interface
+   *    preproccesing in CDriver ---*/
+  if (nMarker_Turbomachinery != 0) {
+    nTurboInterfaces = (nMarker_Turbomachinery -  1)*2; //Two markers per zone minus inlet & outlet
+    Kind_TurboInterface.resize(nTurboInterfaces);
+    /*--- Loop over all markers ---*/
+    for (iMarker_CfgFile = 0; iMarker_CfgFile < nMarker_CfgFile; iMarker_CfgFile++) {
+      /*--- Identify mixing plane markers ---*/
+      if (Marker_MixingPlaneInterface != nullptr){ // Necessary in cases where no mixing plane interfaces are defined
+        if (Marker_CfgFile_MixingPlaneInterface[iMarker_CfgFile] != 0) { //Is a mixing plane
+          /*--- Find which list position this marker is in turbomachinery markers ---*/
+          const auto* target = std::find(&Marker_Turbomachinery[0], &Marker_Turbomachinery[nMarker_Turbomachinery*2-1], Marker_CfgFile_TagBound[iMarker_CfgFile]);
```

Comment:
I take it from the documentation of `nMarkerTurbomachinery` that it is the size of the array `Marker_Turbomachinery`. If this is true, then any memory access beyond `nMarker_Turbomachinery - 1` is illegal, and you'll want to specify `&MarkerTurbomachinery[nMarker_Turbomachinery]` as your _last_ item, i.e., the first item beyond the range to be searched.
```suggestion
          const auto* target = std::find(&Marker_Turbomachinery[0], &Marker_Turbomachinery[nMarker_Turbomachinery], Marker_CfgFile_TagBound[iMarker_CfgFile]);
```
Are you assuming to always find your marker among the turbomachinery markers? If yes, you could make the assumption visible by adding a comment and an assertion in the spirit of
```
assert(target != &Marker_Turbomachinery[nMarker_Turbomachinery]);
```
i.e. test whether the _last_ item was returned. If no, there should be a treatment for not finding the marker.

Reply:
The value of `nMarkerTurbomachinery` is equal to half the size of the array `Marker_Turbomachinery`. See option_structur.inl:1652 `unsigned long nVals = totalVals / mod_num;`. I don't know why it is done like this, but this is used extensively when defining the turbovertex structure and I do not want to go through and change all this. My best guess would be when this was originally implemented it was assumed that each marker specified here will have a corresponding marker in an adjacent blade row where the turboperformance variables don't need to be evaluated again. This pops up everywhere and I think it is more convenient to follow the _convention_ here rather than undertake a much larger refactoring in the other 40 places where this value is used (at least for now). I apply the -1 as I do not want to search for the final value in the MARKER_TURBOMACHINERY array, which will be the outflow.